src/scalarstats.jl

```diff
@@ -260,12 +260,36 @@ variation(x) = ((m, s) = mean_and_std(x); s/m)
 realXcY(x::Real, y::Real) = x*y
 realXcY(x::Complex, y::Complex) = real(x)*real(y) + imag(x)*imag(y)
 
+
+"""
+    var_sample_mean(x[, weights::AbstractWeights]; mean=nothing)
+
+Return the estimated variance of the mean for a collection `x`. When using no weights, this is
+equal to the (sample) standard deviation divided by the sample size. If weights are used, 
+the variance of the sample mean is calculated as follows:
+
+* `AnalyticWeights`: Not implemented.
+* `FrequencyWeights`: ``\\frac{\\sum_{i=1}^n w_i (x_i - \\bar{x_i})^2}{(\\sum w_i) (\\sum w_i - 1)}``
+* `ProbabilityWeights`: ``\\frac{n}{n-1} \\frac{\\sum_{i=1}^n w_i^2 (x_i - \\bar{x_i})^2}{\\left( \\sum w_i \\right)^2}``
```

Comment:
Do you have any references for these formulas?

Reply:
> Do you have any references for these formulas?

Taken off Wikipedia, which seems to give [this source](https://www.sciencedirect.com/science/article/pii/135223109400210C?via%3Dihub). I compared the results to what you get in R's "survey" package, and it's also the estimator used in all the importance sampling literature I've seen.

Reply:
> Do you have any references for these formulas?

Taken off Wikipedia, which seems to give [this source](https://www.sciencedirect.com/science/article/pii/135223109400210C?via%3Dihub). I compared the results to what you get in R's "survey" package, and it's also the estimator used in all the importance sampling literature I've seen.

Reply:
> Do you have any references for these formulas?

Taken off Wikipedia, which seems to give [this source](https://www.sciencedirect.com/science/article/pii/135223109400210C?via%3Dihub). I compared the results to what you get in R's "survey" package, and it's also the estimator used in all the importance sampling literature I've seen.

Reply:
Neither the paper nor Wikipedia seem to discuss if and how these estimators are related to probability or frequency weights. In fact, the paper even mentions that there is no standard and clear definition in the field and uses an uncorrected variance estimate.

Reply:
> Neither the paper nor Wikipedia seem to discuss if and how these estimators are related to probability or frequency weights. In fact, the paper even mentions that there is no standard and clear definition in the field and uses an uncorrected variance estimate.

It seems there's no standard definition when sampling from a finite population, because any estimator will be biased; the Wikipedia article is also a bit of a mess. I found a much better source which deals with the case of sampling with replacement (or from an infinitely-large population); Carl Erik Sarndal, et. al - Model Assisted Survey Sampling (1992), pages 51 and 52. The estimator used here is unbiased when sampling with replacement, when the weights are inverse-probability weights. It's in Wikipedia as well, [here](https://en.wikipedia.org/wiki/Weighted_arithmetic_mean#Variance_of_the_weighted_sum_(pwr-estimator_for_totals)); the relevant section is the part right after the phrase "The above formula was taken from Sarndal et. al. (1992)."

Reply:
As for frequency weights, the weights are discussed [here](https://en.wikipedia.org/wiki/Weighted_arithmetic_mean#Frequency_weights). I don't have a citation for them but the formula is obvious enough that I don't think one would be required.

Reply:
The R survey package is indeed the definitive reference for probability weights. But please add details and references about formulas used to the docstring.

The case of frequency weights is indeed simple as it can always be compared to replicating each observation the number of times corresponding to its weight.